gh-badges/lib/index.js

```diff
@@ -3,85 +3,85 @@
  * @module gh-badges
  */
 
-const makeBadge = require('./make-badge')
+const _makeBadge = require('./make-badge')
 
 class ValidationError extends Error {}
 
-/**
- * BadgeFactory
- */
-class BadgeFactory {
-  constructor(options) {
-    if (options !== undefined) {
-      console.error(
-        'BadgeFactory: Constructor options are deprecated and will be ignored'
-      )
-    }
+function _validate(format) {
+  if (!('text' in format)) {
+    throw new ValidationError('Field `text` is required')
   }
 
-  _validate(format) {
-    if (!('text' in format)) {
-      throw new ValidationError('Field `text` is required')
-    }
+  if (
+    !Array.isArray(format.text) ||
+    format.text.length !== 2 ||
+    typeof format.text[0] !== 'string' ||
+    typeof format.text[1] !== 'string'
+  ) {
+    throw new ValidationError('Field `text` must be an array of 2 strings')
+  }
 
-    if (
-      !Array.isArray(format.text) ||
-      format.text.length !== 2 ||
-      typeof format.text[0] !== 'string' ||
-      typeof format.text[1] !== 'string'
-    ) {
-      throw new ValidationError('Field `text` must be an array of 2 strings')
+  const stringFields = ['labelColor', 'color']
+  stringFields.forEach(function(field) {
+    if (field in format && typeof format[field] !== 'string') {
+      throw new ValidationError(`Field \`${field}\` must be of type string`)
     }
+  })
 
-    const stringFields = ['labelColor', 'color']
-    stringFields.forEach(function(field) {
-      if (field in format && typeof format[field] !== 'string') {
-        throw new ValidationError(`Field \`${field}\` must be of type string`)
-      }
-    })
+  const formatValues = ['svg', 'json']
+  if ('format' in format && !formatValues.includes(format.format)) {
+    throw new ValidationError(
+      `Field \`format\` must be one of (${formatValues.toString()})`
+    )
+  }
 
-    const formatValues = ['svg', 'json']
-    if ('format' in format && !formatValues.includes(format.format)) {
-      throw new ValidationError(
-        `Field \`format\` must be one of (${formatValues.toString()})`
-      )
-    }
+  const templateValues = [
+    'plastic',
+    'flat',
+    'flat-square',
+    'for-the-badge',
+    'popout',
+    'popout-square',
+    'social',
+  ]
+  if ('template' in format && !templateValues.includes(format.template)) {
+    throw new ValidationError(
+      `Field \`template\` must be one of (${templateValues.toString()})`
+    )
+  }
+}
 
-    const templateValues = [
-      'plastic',
-      'flat',
-      'flat-square',
-      'for-the-badge',
-      'popout',
-      'popout-square',
-      'social',
-    ]
-    if ('template' in format && !templateValues.includes(format.template)) {
-      throw new ValidationError(
-        `Field \`template\` must be one of (${templateValues.toString()})`
-      )
+function _clean(format) {
```

Comment:
I've added this additional function here which:

* strips any `null` or `undefined` values
* strips any values we don't want to pass on to the other `makeBadge()` ( in `lib/make-badge` )

(without mutating the input object) before we validate the object and pass it on. Its not really strictly related to the rest of this PR, but I realised I should have done this in the validation PR while I was working on this.